lib/index.js

```diff
@@ -246,6 +249,67 @@ export default {
     return provider
   },
 
+  provideHyperclick(): HyperclickProvider {
+    const provider = {
+      grammarScopes: ['source.js', 'source.js.jsx'],
+      getSuggestionForWord: async (textEditor: TextEditor, text: string, range: Range): Promise<?HyperclickSuggestion> => {
+        const filePath = textEditor.getPath()
+        if (!filePath) {
+          return null
+        }
+
+        const fileDirectory = Path.dirname(filePath)
+        const configFile = await findCachedAsync(fileDirectory, '.flowconfig')
+        if (!configFile) {
+          return null
+        }
+
+        const flowOptions = [
+          'get-def',
+          '--json',
+          filePath,
+          range.start.row + 1,
+          range.start.column + 1,
+        ]
+
+        let result
+        try {
+          result = await exec(await this.getExecutablePath(fileDirectory), flowOptions, {
+            cwd: fileDirectory,
+            ignoreExitCode: true,
+            timeout: 60 * 1000,
+            uniqueKey: 'flow-ide-hyperclick',
+          })
+          if (result === null) {
+            return null
+          }
+        } catch (error) {
+          if (error.message.indexOf(INIT_MESSAGE) !== -1 && configFile) {
+            spawnedServers.add(Path.dirname(configFile))
+          }
+          if (error.message.indexOf(INIT_MESSAGE) !== -1 || error.message.indexOf(RECHECKING_MESSAGE) !== -1) {
+            return provider.getSuggestionForWord(textEditor, text, range)
+          }
+          throw error
+        }
+        const jsonResult = JSON.parse(result)
+
+        return jsonResult.path === ''
+          ? null
+          : {
+            priority: this.hyperclickPriority,
+            range,
+            callback() {
+              atom.workspace.open(jsonResult.path).then((editor) => {
```

Comment:
As a second parameter to workspace.open, can you do something like `{ searchAllPanes: true }`?